app/views/compute_resources_vms/form/vmware/_base.html.erb

```diff
@@ -49,6 +50,14 @@ end %>
                { :disabled => images.empty?, :label => _('Image'), :label_size => "col-md-2" } %>
 </div>
 
+<!-- # BIOS firmware doesn't support virtual TPM. -->
+<div id='efi_features' <%= 'class=hide' if !new_vm || f.object.firmware == 'bios' %>>
```

Comment:
The opposite. We hide this option if it is an existing VM or if the firmware is set to BIOS. 
However, there's an edge case where the firmware is set to Automatic, then the TPM option is available, and if you select it and set the PXE Loader to BIOS, you will get the following error:
`Failed to power up a compute VMware_with_keys (VMware) instance <HOST NAME>: GenericVmConfigFault: The virtual machine must use EFI firmware.`

Reply:
> The opposite. We hide this option if it is an existing VM or if the firmware is set to BIOS.

`f.object.firmware` can only be set on an existing VM. If you always hide it on an existing VM, isn't it redundant? In other words:
```suggestion
<div id='efi_features' <%= 'class=hide' if !new_vm %>>
```

> However, there's an edge case where the firmware is set to Automatic, then the TPM option is available, and if you select it and set the PXE Loader to BIOS, you will get the following error:

This is what I was getting at. At first, I'd catch the situation in code before it's even sent to VMware so you can raise an error in the form. Then you can decide if you want to add some front end code to handle this. It may be useful given the two form controls are on different tabs but I wouldn't block on it when there's a proper handling post-submit.